File: 0495-teemo-attacking/0495-teemo-attacking.rb
# @param {Integer[]} time_series
# @param {Integer} duration
# @return {Integer}
def find_poisoned_duration(time_series, duration)
    time_poisoned = 0
    (time_series.size-1).times do |i| # loopsize -1 (skip last)
        time_poisoned += [duration, (time_series[i+1]-time_series[i])].min
    end
    time_poisoned + duration
end

File: 0495-teemo-attacking/0495-teemo-attacking.cpp
int init=[]{ios_base::sync_with_stdio(false);cin.tie(0);cout.tie(0);return 0;}();
class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        int timePoisoned = 0;
        for (int i=0; i < timeSeries.size() - 1; ++i) {
            timePoisoned += min(duration, (timeSeries[i+1]-timeSeries[i]));
        }
        return timePoisoned + duration;
    }
};
